mocca-client/src/main/java/com/paypal/mocca/client/MoccaSerializer.java

```diff
@@ -369,13 +377,57 @@ private void writeSelectionSet(final ByteArrayOutputStream requestPayload, final
+     *
+     * @param requestPayload the output stream object used to write the request payload, based on the other parameters
+     * @param ignoreFields The list of fields which have to be ignored in selection set generation
+     * @param responseType the return type set in the GraphQL operation method, useful when defining the request selection set
+     */
+    private void writeUserProvidedSelectionSetUsingIgnore(ByteArrayOutputStream requestPayload,
+                                                          List<String> ignoreFields,
+                                                          Type responseType) {
+
+        try {
+            BeanInfo info = Introspector.getBeanInfo(erase(responseType));
+            write(requestPayload, " {");
+
+            final PropertyDescriptor[] pds = info.getPropertyDescriptors();
+
+            final List<String> selectionSet = Arrays.stream(pds)
+                    .filter(pd -> !pd.getName().equals("class"))
+                    .filter(pd -> !ignoreFields.contains(pd.getName()))
+                    .map(pd -> new Tuple<Class<?>>(pd.getName(), pd.getReadMethod().getReturnType()))
+                    .map(e -> {
+                        Class<?> c = e.value;
+                        if (isPojo(c)) {
+                            ByteArrayOutputStream complexVariable = new ByteArrayOutputStream();
+                            final List<String> nextIgnoreFields = getNextIgnoreFields(e.key + ".", ignoreFields);
+                            writeUserProvidedSelectionSetUsingIgnore(complexVariable, nextIgnoreFields, c);
+                            return e.key + complexVariable.toString();
+                        } else {
+                            return e.key;
+                        }
+                    })
+                    .collect(Collectors.toList());
+
+            write(requestPayload, String.join(" ", selectionSet));
+            write(requestPayload, "}");
+        } catch (IntrospectionException | IOException e) {
+            throw new MoccaException("An error happened when writing selection set of type " + responseType, e);
+        }
+    }
+
     /*
-     * Writes the selection set of the GraphQL request message using the user provided SelectionSet annotation as reference.
+     * Writes the selection set of the GraphQL request message using the user provided SelectionSet annotation with Value attribute set as reference.
      *
      * @param requestPayload the output stream object used to write the selection set, based on the other parameters
      * @param selectionSet the SelectionSet annotation set in the GraphQL operation method, necessary to set the selection set
      */
-    private void writeUserProvidedSelectionSet(final ByteArrayOutputStream requestPayload, final SelectionSet selectionSet) {
+    private void writeUserProvidedSelectionSetUsingValue(final ByteArrayOutputStream requestPayload, final SelectionSet selectionSet) {
```

Comment:
Since we are going to reuse `writeResponseTypeSelectionSet`, let's undo this change please. It won't be necessary anymore.